jdaviz/configs/default/plugins/model_fitting/fitting_backend.py

```diff
@@ -355,8 +367,19 @@ def _generate_spaxel_list(spectrum):
     spaxels : list
         List with spaxels
     """
-    spx = [[(x, y) for x in range(spectrum.flux.shape[0])]
-           for y in range(spectrum.flux.shape[1])]
+    if spectrum.mask is None:
+        # If there is no mask, make one:
+        mask = np.zeros(spectrum.shape).astype(bool)
+    elif spectrum.mask.shape != spectrum.flux.shape:
+        # If the mask isn't the right shape, broadcast it:
+        mask = np.broadcast_to(spectrum.mask, spectrum.flux.shape)
+    else:
+        mask = spectrum.mask
+
+    # return only non-masked spaxels
+    n_x, n_y, _ = spectrum.flux.shape
+    spx = [[(x, y) for x in range(n_x) if np.any(mask[x, y])]
+           for y in range(n_y) if np.any(mask[:, y])]
```

Comment:
```suggestion
    spx = [[(x, y) for x in range(n_x) if np.any(~mask[x, y])]
           for y in range(n_y) if np.any(~mask[:, y])]
```

I _think_ that the way you currently have it gives the opposite of what we want - `np.any` will return if any pixels in the spaxel `are` masked (since True=masked), so a fully-masked spaxel would be returned here while a fully non-masked spaxel would be not.

Reply:
Yikes, thanks for this catch. I have a bad habit of making `mask`s in my work (`True`==good) that have the opposite meaning of the numpy convention (`True`==bad). I should stop it but it's a weird habit to break.

Reply:
For reasons I'll never know, it made more sense to novice-numpy-user me to write `plt.plot(x[mask], y[mask])` without needing a logical not. 🤷🏻‍♂️